server/src/test/java/org/elasticsearch/index/query/BoolQueryBuilderTests.java

```diff
@@ -447,4 +450,17 @@ public void testMustRewrite() throws IOException {
                 () -> boolQuery.toQuery(context));
         assertEquals("Rewrite first", e.getMessage());
     }
+
+    public void testExceedMaxNestedDepth() throws IOException {
+        BoolQueryBuilder query = new BoolQueryBuilder();
+        query.should(new BoolQueryBuilder().should(new BoolQueryBuilder().should(RandomQueryBuilder.createQuery(random()))));
+        BoolQueryBuilder.setMaxNestedDepth(2);
+        try (XContentParser parser = createParser(JsonXContent.jsonXContent, query.toString())) {
+            XContentParseException e = expectThrows(XContentParseException.class,
+                () -> parseInnerQueryBuilder(parser));
+            assertThat(e.getCause().getCause(), Matchers.instanceOf(IllegalArgumentException.class));
+            assertEquals(e.getCause().getCause().getMessage(), "The nested depth of the query exceeds the maximum nested depth for bool queries set in [" +
+                INDICES_MAX_NESTED_DEPTH_SETTING.getKey() + "]");
```

Comment:
The format of `assertEquals` is `assertEquals(expected, actual)`, so we should write:
```java
assertEquals("The nested...", e.getCause().getCause().getMessage());
```